velox/dwio/common/ColumnVisitors.h

```diff
@@ -1406,14 +1402,33 @@ class ExtractStringDictionaryToGenericHook {
 
  private:
   int32_t dictionarySize() const {
-    return state_.dictionary.numValues;
+    return state_->dictionary.numValues;
   }
 
   ValueHook* const hook_;
   RowSet const rows_;
-  RawScanState state_;
+  RawScanState* state_;
```

Comment:
This is a lightweight object and ok to copy on each batch.  Just leave it copied to be consistent with other visitors.

Reply:
The `RawScanState` object cannot be copied because for Parquet, when `ExtractStringDictionaryToGenericHook` is defined, `RawScanState#dictionary` is not defined yet. If the object is copied, the subsequent `state_->dictionary.values` ​​operation will fail.

Reply:
Is this still the case now with `ParquetData::hasDictionary_` added?

Reply:
This will still fail.  `RawScanState#dictionary` is initialized when reading datapage later.